types/hapi-auth-basic/index.d.ts

```diff
@@ -1,4 +1,4 @@
-// Type definitions for hapi-bauth-basic 5.0.0
+// Type definitions for hapi-auth-basic 5.0.0
```

Comment:
As part of this PR, could you make the following changes:
- make the version string just `5.0` (patch version is not supposed to be there)
- in the tslint.json in this folder, remove the line that is disabling the `dtheader` rule.

It would be good to remove all rule disabling lines from the tslint.json, but that seems outside the scope of this PR.